karton/core/main.py

```diff
@@ -1,11 +1,125 @@
+
 from .__version__ import __version__
 from .backend import KartonBackend
 from .config import Config
 from .karton import Consumer
 
+log = logging.getLogger(__name__)
+
+
+def get_user_option(prompt: str, default: str) -> str:
+    user_input = input(prompt + f"\n[{default}] ")
+    print("")  # just for style
+    return user_input.strip() or default
+
+
+def configuration_wizard(config_filename: str) -> None:
+    config = ConfigParser()
+
+    log.info("Configuring MinIO")
+    while True:
+        minio_access_key = get_user_option(
+            "Enter the MinIO access key", default="minioadmin"
+        )
+        minio_secret_key = get_user_option(
+            "Enter the MinIO secret key", default="minioadmin"
+        )
+        minio_address = get_user_option(
+            "Enter the MinIO address", default="localhost:9000"
+        )
+        minio_bucket = get_user_option(
+            "Enter the MinIO bucket to use", default="karton"
+        )
+        minio_secure = get_user_option('Use SSL ("0", "1")?', default="0")
+
+        log.info("Testing MinIO connection...")
+        minio = Minio(
+            endpoint=minio_address,
+            access_key=minio_access_key,
+            secret_key=minio_secret_key,
+            secure=bool(int(minio_secure)),
+        )
+        bucket_exists = False
+        try:
+            bucket_exists = minio.bucket_exists(minio_bucket)
+        except Exception as e:
+            log.info("Error while connecting to MinIO: %s", e)
+            retry = get_user_option(
+                'Do you want to try with different MinIO settings ("yes", "no")?',
+                default="yes",
+            )
+            if retry != "yes":
+                log.info("Quitting configuration")
+                return
+            else:
+                continue
+
+        log.info("Connected to MinIO successfully")
+        if not bucket_exists:
+            log.info(
+                (
+                    "The required bucket %s does not exist. To create it automatically,"
+                    " start karton-system with --setup-bucket flag"
+                ),
+                minio_bucket,
+            )
+        break
+
+    config["minio"] = {
+        "access_key": minio_access_key,
+        "secret_key": minio_secret_key,
+        "address": minio_address,
+        "bucket": minio_bucket,
+        "secure": str(bool(int(minio_secure))),
+    }
+
+    log.info("Configuring Redis")
+
+    while True:
+        redis_host = get_user_option("Enter the Redis host", default="localhost")
+        redis_port = get_user_option("Enter the Redis port", default="6379")
+
+        log.info("Testing Redis connection...")
+        redis = StrictRedis(
+            host=redis_host,
+            port=int(redis_port),
+            decode_responses=True,
+        )
+        try:
+            redis.ping()
+        except Exception as e:
+            log.info("Error while connecting to Redis: %s", e)
```

Comment:
If accepted the previous suggestion:
```suggestion
            log.info("Error while connecting to Redis: %s", e. exc_info=True)
```